### PR TITLE
Chore/refine iterations first page

### LINKED ISSUE #44: Implement iteration dashboard with recommendations
After finishing the questionnaire we want to show the user the results and list all recommendations.

### PR DESCRIPTION
**Pull request by lewisdylan**
Redirected completed iterations to recommendation page.

Closes #44

### CODE REVIEW

**Comment by fsmanuel** [comment]
@lewisdylan take a look at the failing tests and request a review when you fixed the linting errors.

File: app/javascript/component/cards/IterationCard.vue [hunk 1]
```diff
@@ -19,11 +19,24 @@
...
+      <button
+        v-if="iteration.completion_rate < 100"
+        class="btn bg-primary py-1 text-xs mx-auto text-white w-100 rounded-sm"
+        @click="openIteration(iteration)"
+      >
         Continue
       </button>
+
+      <button
+        v-if="iteration.completion_rate == 100"
```
Comment (fsmanuel) [inline app/javascript/component/cards/IterationCard.vue]: In JS you should use `===`

Comment (fsmanuel) [reply app/javascript/component/cards/IterationCard.vue]: I would suggest we change this a bit:
```html
<button
  class="btn py-1 text-xs mx-auto text-white w-100 rounded-sm"
  v-bind:class="[isComplete ?  'bg-red' : 'bg-primary']"
  @click="handleClick(iteration)"
>
  {{ buttonLabel }}
</button>
```

```js
computed: {
  isComplete: function() {
    return iteration.completion_rate === 100;
  },
  buttonLabel: function() {
    return this.isComplete ? 'View Recommendations' : 'Continue';
  }
},
methods: {
  handleClick: function(iteration) {
    let options;

    if (this.isComplete) {
      options = {
        name: 'recommendations-index',
        params: { iteration_id: iteration.id }
      };
    } else {
      options = {
        name: 'question',
        params: {
          id: iteration.starting_question_id,
          iteration_id: iteration.id
        }
      };
    }

    this.$router.push(options);
  }
}
```

File: app/javascript/component/cards/IterationCard.vue [hunk 2]
```diff
@@ -19,11 +19,24 @@
       </h1>
 
       <h3 class="text-xs text-weight-light">
-        Started: 28 Jan 2019
+        {{ iteration.created_at }}
```
Comment (fsmanuel) [inline app/javascript/component/cards/IterationCard.vue]: I think we need to change the format of the datetime (parse it as a date) so I would make it a computed property so we can change the format there.

File: app/javascript/views/recommendations/Index.vue [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+    </div>
+    <div class="w-100 h-1 bg-primary" />
+    <div
+      v-if="recommendations.length"
+      class="w-90 mx-auto pt-4 pb-5"
+    >
+      <div class="flex flex-wrap">
+        <recommendation-card
+          v-for="(recommendation, index) in recommendations"
+          :key="index"
```
Comment (fsmanuel) [inline app/javascript/views/recommendations/Index.vue]: I think the `recommendation.id` is a better `key`

File: app/javascript/component/cards/IterationCard.vue (hunk 1 above)
Comment (lewisdylan) [reply app/javascript/component/cards/IterationCard.vue]: Thanks that's better

Comment (musayann) [reply app/javascript/component/cards/IterationCard.vue]: We have already the `completed?` attribute in `Iteration`, I think we should use that one instead

**Comment by musayann** [comment]
Should we merge this one @fsmanuel  ??

File: app/javascript/component/cards/IterationCard.vue [hunk 4]
```diff
@@ -36,6 +40,39 @@ export default {
...
     }
-  }
+  },
+  computed: {
+    startedOn() {
+      let date = new Date(this.iteration.created_at);
+      return date.getDate() + '-' + (date.getMonth() + 1) + '-' + date.getFullYear();
+    },
+    isComplete() {
+      return this.iteration.completion_rate == 100
```
Comment (fsmanuel) [inline app/javascript/component/cards/IterationCard.vue]: `===`

**Pull request merged**